### PR TITLE
Add codeql code checking

### PR DESCRIPTION
**Pull request by MarcelMue**

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply .github/workflows/codeql.yaml:27]: Not sure yet - I have only taken this from upstream.

**Review by yulianedyalkova — APPROVED** [review]

**Comment by MarcelMue** [comment]
Forgot to tell you @stone-z I tried this out - no findings, so not sure if that's a good thing :D

It takes very long to run CodeQL, so not sure if it's worth it.

**Comment by stone-z** [comment]
> It takes very long to run CodeQL, so not sure if it's worth it.

Yeah that is a long time, CodeQL sounds interesting to try though. Maybe if we didn't run on every PR commit. Only release PRs, maybe?

**Comment by tfussell** [comment]
What do you want to do with this one @MarcelMue?

**Comment by MarcelMue** [comment]
Let's close this here. I do not see a big payoff because the runtime is so long.

**Pull request closed without merging**